### DISCUSSION TITLE
network share disk space usage

### DISCUSSION THREAD

**Discussion by fifo1488** [marcone/teslausb#792] [Q&A] [answered]
I have my teslausb setup via CIFS shared with my NAS. The NAS folder has a disk quota of 100GB. Is there a way for teslausb to removed older recordings from the shared folder in order to have enough disk space to upload newer recordings?

**Comment by KrzysztofHajdamowicz** [comment]
That's a great idea!

What comes to my mind is to use find command to delete directories without .nodelete or .preserve file inside.  
I can create some oneliner for that in my spare time (hopefully next week)

**Reply by KrzysztofHajdamowicz** [reply]
Some starting point for further development: https://chat.openai.com/share/dd6cbfed-6e00-49cd-bc07-50fcff161ad9

One thing that smells fishy for me is `sh -c [...]; sh {}` thing.

**Reply by KrzysztofHajdamowicz** [reply]
Looks like GPT-3.5 did it slightly less fishy: https://chat.openai.com/c/3ed4f0a5-d13b-4db3-a9f3-98236dc08b82

**Comment by marcone** [comment]
Only 100GB? You can keep that much on the Pi itself.
In any case, I would suggest managing this on the NAS, not with TeslaUSB.

**Comment by mrd0n** [comment, marked as answer]
Just sharing my approach to cleaning out the archive target.

I have these commands running once a day in the crontab of the system that teslausb rsync’s with:

/usr/bin/find /mnt/TeslaCam/SentryClips/ -mtime +45 | xargs rm -rf {}
/usr/bin/find /mnt/TeslaCam/SentryClips -type d -empty -delete
/usr/bin/find /mnt/TeslaCam/SavedClips/ -mtime +45 | xargs rm -rf {}
/usr/bin/find /mnt/TeslaCam/SavedClips -type d -empty -delete

The first line returns all the SentryClips files older than 45 days and deletes them
The second line removes any empty SentryClips folders
The third and forth line do the same for SavedClips

It’s not a one-liner, but simple enough to implement.
Love the project.

**Reply by marcone** [reply]
This is a good example of why I think this should be managed on the NAS, not by TeslaUSB. There are many variables and edge cases to consider:
- how long to keep things?
- what if the user wants different expiration times for Sentry, Saved and Recent?
- what if the user wants to keep some files forever (e.g. like the ".nodelete" example given above)
- what if after applying all the specified expiration rules there still isn't enough free space to archive a new batch of files?

**Comment by alexmereness** [comment]
> Just sharing my approach to cleaning out the archive target. I have these commands running once a day in the crontab of the system that teslausb rsync’s with: /usr/bin/find /mnt/TeslaCam/SentryClips/ -mtime +45 | xargs rm -rf {} /usr/bin/find /mnt/TeslaCam/SentryClips -type d -empty -delete /usr/bin/find /mnt/TeslaCam/SavedClips/ -mtime +45 | xargs rm -rf {} /usr/bin/find /mnt/TeslaCam/SavedClips -type d -empty -delete The first line returns all the SentryClips files older than 45 days and deletes them The second line removes any empty SentryClips folders The third and forth line do the same for SavedClips It’s not a one-liner, but simple enough to implement. Love the project.
> […](#)
> On Oct 18, 2023, at 11:35 AM, KrzysztofHajdamowicz ***@***.***> wrote: Looks like GPT-3.5 did it slightly less fishy: https://chat.openai.com/c/3ed4f0a5-d13b-4db3-a9f3-98236dc08b82 — Reply to this email directly, view it on GitHub <[#792 (reply in thread)](https://github.com/marcone/teslausb/discussions/792#discussioncomment-7318364)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AO3B6UPN2MYTSDRHJA6FFO3YAAHM7AVCNFSM6AAAAAA6FWVUHCVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM3TGMJYGM3DI>. You are receiving this because you are subscribed to this thread.

So, what's your contab file look like? Are you running it on a timed schedule? At boot? I generally agree that the ideal place to manage this would be on the NAS. My concern is the inconsistency in my Pi being on the network.

**Reply by mrd0n** [reply]
My crontab setup for the target system that teslaUSB syncs with runs the archive purge every day starting at 2:00am.   This is running on a different linux host than the TeslaUSB Raspberry pi, so it doesn't require the Raspberry Pi to be online.  Some NAS systems have a built-in cron scheduling service, negating the need for another box or to have this run on the raspberry pi within the car.

Here is what my crontab looks like:
```
# m h  dom mon dow   command
00 2 * * * /usr/bin/find /mnt/Drobo5N-teslausb/TeslaCam/SentryClips/ -mtime +45 | xargs rm -rf {}
30 2 * * * /usr/bin/find /mnt/Drobo5N-teslausb/TeslaCam/SavedClips/ -mtime +45 | xargs rm -rf {}
30 4 * * * /usr/bin/find /mnt/Drobo5N-teslausb/TeslaCam/SavedClips -type d -empty -delete
30 4 * * * /usr/bin/find /mnt/Drobo5N-teslausb/TeslaCam/SentryClips -type d -empty -delete
```

**Comment by alexmereness** [comment]
For others looking for another option, I created a Windows PowerShell script that can be used manage (delete) files. I did a write up on it here: https://github.com/marcone/teslausb/discussions/824
